### PR TITLE
2016 01 06 adhocracy integration

### PR DESCRIPTION
**Pull request by libscott**
Still needs to take into consideration:

* Copying config files
* Copying correct binary
* How to organise multiple separate release scripts

Related: https://github.com/ssadler/adhocracy3/commit/5c73b4742edf16a4a46c2685ba7715ad2b1db79e

### CODE REVIEW

File: misc/release.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
```
Comment (fisx) [inline misc/release.sh]: (some day i want to move some of the stuff from `misc` to maybe `bin` or `scripts`.  but not necessarily now i guess.)

File: misc/release.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+# release.sh
+#
+# Build a Thentos binary release tarball.
+# Must be run from the base of the thentos project,
```
Comment (fisx) [inline misc/release.sh]: https://github.com/liqd/thentos/blob/master/thentos-purescript/build.sh#L4 shows how you can make this more robust.

File: misc/release.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+cp $cabal_sandbox/bin/thentos-adhocracy . # TODO: copy correct binary(s)
+mkdir -p thentos-core/schema
+cp -r ../thentos/thentos-core/schema/* thentos-core/schema
+#TODO: Copy config
+cd ..
+tar cf $relname.tar.gz *
+popd
+cp $tmpdir/$relname.tar.gz .
+rm -rf $tmpdir
+md5sum $relname.tar.gz
```
Comment (fisx) [inline misc/release.sh]: ...  `> $relname.SHA1`?

**Comment by fisx** [comment]
looks good.  much simpler than my egg magic.  (:

i think the question of having several parallel releases (thentos-captcha vs. thentos-proxy) is related to the question of having several binaries, right?  so it can be postponed a bit.  (my feeling is that i would like a 1:1 relationship between release scripts, binaries, and tar balls.)

**Comment by ssadler** [comment]
Release example: https://github.com/ssadler/adhocracy3/releases/tag/test

File: misc/release.sh (hunk 2 above)
Comment (ssadler) [inline misc/release.sh]: Sure

**Comment by ssadler** [comment]
> (my feeling is that i would like a 1:1 relationship between release scripts, binaries, and tar balls.)

+1

**Comment by fisx** [comment]
I removed the 'WORK IN PROGRESS' from the description and will take a look now.

File: misc/\ [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
```
Comment (fisx) [inline misc/\]: bogus git-add

File: misc/release/thentos-captcha-README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+Web service that serves captcha images and sounds and verifies captcha solutions.
+
+## Install
+
+    apt-get install postgresql-client espeak libgmp10
+
+## Configure
+
+The relavent sections in `thentos-captcha.config` are: `frontend`, `backend`, `log`, and `database.
```
Comment (fisx) [inline misc/release/thentos-captcha-README.md]: missing quote.

File: thentos-core/mkcap.hs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (fisx) [inline thentos-core/mkcap.hs]: bogus git-add

File: thentos-core/thentos-core.cabal [hunk 7]
```diff
@@ -1,5 +1,5 @@
 name:                thentos-core
-version:             0.0.1.1
+version:             0.1.0
```
Comment (fisx) [inline thentos-core/thentos-core.cabal]: this breaks the install script, as the other packages expect version 0.0.1.1

**Comment by fisx** [comment]
The v0.1.0 contains absolute paths from the build machine.  I fixed that in #448, but found that the binary still references the so files from the sandbox, even if data files are now all relative to `.`.  I don't see anything in this PR that deals with this, so I'll go and do some research myself on static builds.

**Comment by fisx** [comment]
closing in favour of #453

**Pull request closed without merging**